stubmodify/modify_test.go

```diff
@@ -78,13 +78,46 @@ func TestWriteAttributionCodeFull(t *testing.T) {
 			t.Error("fileBytes was modified in WriteAttributionCode")
 		}
 
+		// Ensure the size didn't change
 		if len(modBytes) != len(fileBytes) {
 			t.Errorf("modBytes: %d != fileBytes: %d", modBytes, fileBytes)
 		}
 
+		// Ensure the attribution code is included in the modified bytes
 		if !bytes.Contains(modBytes, code) {
 			t.Errorf("modBytes does not contain code: %v", code)
 		}
+
+		// This is not as robust as the way WriteAttributionCode finds the offset
+		// but it is likely good enough for tests -- and at the very least, means
+		// that we will not have false negatives or positives if that implementation
+		// is broken. Note that we use LastIndex because `MozTag` may be present
```

Comment:
Do we have tests for that part?

Reply:
A test for the tests? :)

I mean, I guess I could....but I think the real alternative here is what I did before (returning the offset that WriteAttributionData found), but I'm not sure that's a great option?

I'm happy to do whatever you think is best is here.

Reply:
Sorry, I meant the part about the presence of the moz tag multiple times. The new test code here looks good but the thing about the moz tag being present more than once is new to me.

Reply:
Ooooh. No, there's no test for that. I don't _think_ there needs to be, because I don't think the first part is part of the signature area:
```
hexdump -C testdata/test-stub.exe | grep MOZ   
00009fd0  15 2d d1 96 fc ff 0e d8  5f 5f 4d 4f 5a 43 55 53  |.-......__MOZCUS|
00039c40  04 82 04 00 5f 5f 4d 4f  5a 43 55 53 54 4f 4d 5f  |....__MOZCUSTOM_|
```

I'm honestly not sure _exactly_ why there's two, and why it's only present in certain files. It's possible that this is simply a reflection of [the code that we embed that deals with extracting attribution information](https://searchfox.org/mozilla-central/source/other-licenses/7zstub/src/CPP/7zip/Bundles/SFXSetup/SfxSetup.cpp#182) -- but I haven't confirmed that.

Reply:
I see.

The fact that we need to pick the last occurrence is a bit fragile. I suppose it works because the certificate table is near the end? Ha well, I suppose this is fine for now.

Reply:
I can see why it could be fragile, but I think this is actually fine, because the way `WriteAttributionCode` finds the location is by parsing the signature / certificate table. I'm quite certain that the first occurence is not part of that, so there's really no chance we'll ever end up there. (This might be why we do it this why....I really have no idea at this point.)